qa/src/test/resources/features/datastore/Datastore.feature

```diff
@@ -331,6 +331,270 @@ Feature: Datastore tests
+      |metric        |type      |value      |
+      |tst-metric-5  |string    |123        |
+      |tst-metric-6  |bool      |true       |
+    Then I store the messages from list "TestMessages" and remember the IDs as "StoredMessageIDs"
+    Then I store the messages from list "TestMessages2" and remember the IDs as "StoredMessageIDs2"
+    And I refresh all database indices
+    When I query for the metrics from client "test-client-1" and store them as "Client1Metrics"
+    Then There are exactly 4 metrics in the list "Client1Metrics"
+    And The metric info items "Client1Metrics" match the prepared messages in "TestMessages"
+    And All indices are deleted
+
+  Scenario: MetricsInfo last published date
+    Check the correctness of the metrics info last publish date stored by retrieving the
+    metrics info published in a defined time window.
+
+    Given I login as user with name "kapua-sys" and password "kapua-password"
+    And Account for "kapua-sys"
+    And The device "test-device-1"
+    And I set the database to device timestamp indexing
+    When I prepare a number of messages with the following details and remember the list as "TestMessages"
+      |clientId      |topic              |captured            |
+      |test-client-1 |test_topic/1/2/3   |03/07/2017T09:00:00 |
+      |test-client-2 |test_topic/1/2/3   |03/07/2017T09:00:00 |
+      |test-client-2 |test_topic/1/2/3   |03/07/2017T09:00:10 |
+      |test-client-2 |test_topic/1/2/3   |03/07/2017T09:00:20 |
+    And I set the following metrics to the message 0 from the list "TestMessages"
+      |metric        |type      |value      |
+      |tst-metric-1  |double    |123        |
+      |tst-metric-2  |int       |123        |
+    And I set the following metrics to the message 1 from the list "TestMessages"
+      |metric        |type      |value      |
+      |tst-metric-3  |string    |123        |
+      |tst-metric-4  |bool      |true       |
+    And I set the following metrics to the message 2 from the list "TestMessages"
+      |metric        |type      |value      |
+      |tst-metric-3  |string    |123        |
+      |tst-metric-4  |bool      |true       |
+    And I set the following metrics to the message 3 from the list "TestMessages"
+      |metric        |type      |value      |
+      |tst-metric-3  |string    |123        |
+      |tst-metric-4  |bool      |true       |
+    Then I store the messages from list "TestMessages" and remember the IDs as "StoredMessageIDs"
+    And I refresh all database indices
+    When I query for the current account metrics in the date range "03/07/2017T09:00:00" to "03/07/2017T09:00:20" and store them as "AccountMetrics"
+    Then There are exactly 4 metrics in the list "AccountMetrics"
+    And The metric info items "AccountMetrics" match the prepared messages in "TestMessages"
+    And Client "test-client-1" first published a metric in the list "AccountMetrics" on "03/07/2017T09:00:00"
```

Comment:
may we miss two conditions around this line (see https://github.com/eclipse/kapua/blob/develop/service/datastore/internal/src/test/java/org/eclipse/kapua/service/datastore/internal/MessageStoreServiceTest.java#L1070)

    And Client "test-client-1" last published a metric in the list "AccountMetrics" on "03/07/2017T09:00:00"
    And Client "test-client-2" first published a metric in the list "AccountMetrics" on "03/07/2017T09:00:00"